### PR TITLE
fix(GCS+gRPC): conflict with Win32 global function

### PR DESCRIPTION
**Pull request by coryan**
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/12522)

### CODE REVIEW

File: google/cloud/storage/examples/storage_async_samples.cc [hunk 1]
```diff
@@ -140,8 +140,10 @@ void InsertObjectVectorVectors(
   (client, argv.at(0), argv.at(1));
 }
 
-void DeleteObject(google::cloud::storage_experimental::AsyncClient& client,
-                  std::vector<std::string> const& argv) {
+// The name is redundant, but worksaround a conflict with the global
```
Comment (dbolduc) [inline google/cloud/storage/examples/storage_async_samples.cc:143]: works around

**Review by dbolduc — APPROVED** [review]

File: google/cloud/storage/examples/storage_async_samples.cc (hunk 1 above)
Comment (coryan) [reply google/cloud/storage/examples/storage_async_samples.cc]: Fixed and merged

File: google/cloud/storage/examples/storage_async_samples.cc [hunk 2]
```diff
@@ -140,8 +140,10 @@ void InsertObjectVectorVectors(
   (client, argv.at(0), argv.at(1));
 }
 
-void DeleteObject(google::cloud::storage_experimental::AsyncClient& client,
-                  std::vector<std::string> const& argv) {
+// The name is redundant, but works around a conflict with the global
```
Comment (devbww) [inline google/cloud/storage/examples/storage_async_samples.cc:143]: "The name is arbitrary" perhaps?

Comment (coryan) [reply google/cloud/storage/examples/storage_async_samples.cc:143]: The `Async*` part of the name is redundant in a file called `storage_async_samples.cc`. The file name is also repetitive: google/cloud/**storage**/examples/**storage**_async_samples.cc, but I digress).

But the intent is obviously not clear, maybe "Calling this function `Async*` seems repetitive given the file name"?

Comment (devbww) [reply google/cloud/storage/examples/storage_async_samples.cc:143]: Don't listen to me too much, but "the name is redundant" implies that the name could be removed.  So "the `Async` part of the name is redundant" would be better, yes.

But it sounds like you're really saying is, "We'd like to call this just `DeleteObject()`, but that conflicts with the name of a Windows function."  (Presumably it is actually a macro given that this is in an anonymous namespace.)

Anyway ... not a big deal of course.  Forget about it if the argument isn't compelling enough to warrant a change.

**Pull request merged**